component/src/main/java/org/wso2/extension/siddhi/execution/json/JsonTokenizerStreamProcessorFunction.java

```diff
@@ -0,0 +1,204 @@
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.exception.SiddhiAppValidationException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class provides implementation for tokenizing the given json based on a specific path.
+ */
+@Extension(
+        name = "tokenize",
+        namespace = "json",
+        description = "This will tokenize the given json according the path provided",
+        parameters = {
+                @Parameter(
+                        name = "json",
+                        description = "The input json which should be tokenized using the given path.",
+                        type = {DataType.STRING, DataType.OBJECT}),
+                @Parameter(
+                        name = "path",
+                        description = "The path which is used to tokenize the given json",
+                        type = {DataType.STRING})
+        },
+        returnAttributes = {
+                @ReturnAttribute(
+                        name = "jsonElement",
+                        description = "The json element retrieved based on the given path and the json",
+                        type = {DataType.STRING})},
+        examples = @Example(
+                syntax = "define stream InputStream (json string,path string);\n" +
+                        "@info(name = 'query1')\n" +
+                        "from InputStream#json:tokenize(json, path)\n" +
+                        "select jsonElement\n" +
+                        "insert into OutputStream;",
+                description = "This query performs tokenization for the given json using the path specified.")
+)
+public class JsonTokenizerStreamProcessorFunction extends StreamProcessor {
+    private static final Logger log = Logger.getLogger(JsonTokenizerStreamProcessorFunction.class);
+    private static final Gson gson = new Gson();
+
+    @Override
+    protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processor nextProcessor,
+                           StreamEventCloner streamEventCloner, ComplexEventPopulater complexEventPopulater) {
+        while (streamEventChunk.hasNext()) {
+            StreamEvent streamEvent = streamEventChunk.next();
+            Object jsonInput = attributeExpressionExecutors[0].execute(streamEvent);
+            String path = (String) attributeExpressionExecutors[1].execute(streamEvent);
+            Object filteredJsonElements;
+            try {
+                if (jsonInput instanceof String) {
+                    filteredJsonElements = JsonPath.read((String) jsonInput, path);
+                } else {
+                    filteredJsonElements = JsonPath.read(jsonInput, path);
+                }
+            } catch (PathNotFoundException e) {
+                filteredJsonElements = null;
+                log.warn("Cannot find json element for the path '" + path + "'");
```

Comment:
Add the Siddhi app name, json and path in the log

Reply:
Are we dropping the message or sending null. which one will be useful ? 
Lets get another parameter to allow null or not use the same parameter named used in json-input-mapper